src/lightning/app/cli/connect/maverick.py

```diff
@@ -194,6 +211,55 @@ def connect_maverick(name: str) -> None:
 def disconnect_maverick(name: str) -> None:
     # disconnect stop and remove the docker containers
     with Live(Spinner("point", text=Text("disconnecting maverick...", style="white")), transient=True):
+        try:
+            deregister_from_cloud(name)
+        except Exception as e:
+            rich.print(f"[red]Failed[/red]: Disconnecting machine failed with error: {e}")
+            return
         subprocess.run(f"docker stop {CODE_SERVER_CONTAINER}", shell=True, capture_output=True)
         subprocess.run(f"docker stop {LIGHTNING_DAEMON_CONTAINER}", shell=True, capture_output=True)
     rich.print(f"[green]Succeeded[/green]: maverick {name} has been disconnected from lightning.")
+
+
+def register_to_cloud(name: str, project_name: str) -> None:
+    client = LightningClient(retry=False)
+    projects = client.projects_service_list_memberships()
+    if project_name:
+        for project in projects.memberships:
+            if project.name == project_name:
+                project_id = project.project_id
+                break
+        else:
+            raise ValueError(f"Project {project_name} does not exist.")
+    else:
+        project_id = _get_project(client, verbose=False).project_id
+
+    cluster_bindings = client.projects_service_list_project_cluster_bindings(project_id=project_id)
+    for c in cluster_bindings.clusters:
+        if c.cluster_name == name:
+            existing_cluster = client.cluster_service_get_cluster(id=c.cluster_id)
+            if existing_cluster.status.phase == V1ClusterState.RUNNING:
+                raise RuntimeError(f"Cluster {name} already exists and is running.")
```

Comment:
Let's not raise RuntimeError in a CLI. Let's use the error_and_exit helper.

Reply:
This is being caught by the caller and use `rich.print` to pass the message to the user